dbms/src/DataStreams/WindowTransformAction.h

```diff
@@ -204,7 +205,19 @@ struct WindowTransformAction
             if (ws.cached_block_number != block_number)
             {
                 for (size_t i = 0; i < ws.arguments.size(); ++i)
-                    ws.argument_columns[i] = block.input_columns[ws.arguments[i]].get();
+                {
+                    const IColumn * col = block.input_columns[ws.arguments[i]].get();
+                    if unlikely (col->isColumnConst())
+                    {
+                        ColumnPtr converted = col->convertToFullColumnIfConst();
+
+                        // Put column pointer into materialized_columns to avoid the release of pointer
+                        ws.materialized_columns.push_back(converted);
+                        col = ws.materialized_columns.back().get();
+                    }
+
+                    ws.argument_columns[i] = col;
```

Comment:
since `ws.argument_columns` is per block, I think `ws.materialized_columns` should also be per block

Reply:
> since `ws.argument_columns` is per block, I think `ws.materialized_columns` should also be per block

okk, I move the `ws.materialized_columns.resize(0);`.